src/editor.c

```diff
@@ -2216,7 +2294,24 @@ gboolean editor_start_auto_complete(GeanyEditor *editor, gint pos, gboolean forc
 	root = cword;
 	rootlen = strlen(root);
 
-	if (!ret && rootlen > 0)
+	if (ret || force)
+	{
+		if (autocomplete_scope_shown)
+		{
+			autocomplete_scope_shown = FALSE;
+			if (!ret)
+				sci_send_command(sci, SCI_AUTOCCANCEL);
+		}
+	}
+	else
+	{
+		ret = autocomplete_scope(editor, root, rootlen);
+		if (!ret && autocomplete_scope_shown)
+			sci_send_command(sci, SCI_AUTOCCANCEL);
+		autocomplete_scope_shown = ret;
+	}
+
+	if (!ret && rootlen > 0 && !autocomplete_scope_shown)
```

Comment:
why add the `&& !autocomplete_scope_shown`?  shouldn't `ret` always be `TRUE` in this case?